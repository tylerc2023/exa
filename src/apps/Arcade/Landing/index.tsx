import {useNavigation} from '@react-navigation/native';
import React, {memo, useCallback} from 'react';
import {Button} from 'react-native';
import {Screen} from '../../../components';
import {useAdminNavBack} from '../../../hooks';
import {arcadeScreens} from '../types';

export const Landing = memo(function Landing() {
  const {navigate} = useNavigation();
  const {onLeftPress} = useAdminNavBack();
  const onPress = useCallback((screen: string) => () => navigate(screen), [
    navigate,
  ]);

  return (
    <Screen onLeftPress={onLeftPress} title="Arcade">
      {arcadeScreens.map(screen => (
        <Button key={screen} onPress={onPress(screen)} title={screen} />
      ))}
    </Screen>
  );
});
