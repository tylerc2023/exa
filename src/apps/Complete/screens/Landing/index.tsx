import {useNavigation} from '@react-navigation/native';
import React, {memo, useCallback} from 'react';
import {Button, Carousel, Screen} from '../../../../components';
import {Slide} from '../../../../components/Carousel/types';
import {useColor} from '../../../../hooks';

export const Landing = memo(function Landing() {
  const {navigate} = useNavigation();
  const color = useColor();
  const navPortfolioLogin = useCallback(() => navigate('account'), [navigate]);

  const slides: Slide[] = [
    {id: '1', text: 'hello', color: color.info},
    {id: '2', text: 'bob', color: color.warning},
    {id: '3', text: 'steve', color: color.success},
    {id: '4', text: 'jill', color: color.brand},
  ];

  return (
    <Screen title="Landing">
      <Carousel slides={slides} />
      <Button center onPress={navPortfolioLogin} title="Login" />
    </Screen>
  );
});
