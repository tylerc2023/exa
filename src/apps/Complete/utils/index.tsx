export * from './configs';
export * from './useKeyboardHeight';
export * from './userFactory';
