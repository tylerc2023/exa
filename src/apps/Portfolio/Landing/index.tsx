import {useNavigation} from '@react-navigation/native';
import React, {memo, useCallback} from 'react';
import {Button, Carousel, Screen} from '../../../components';
import {Slide} from '../../../components/Carousel/types';
import {useColor} from '../../../hooks';

export const Landing = memo(function PortfolioLanding() {
  const {navigate, goBack} = useNavigation();
  const color = useColor();
  const navPortfolioLogin = useCallback(() => navigate('login'), [navigate]);
  const navBack = useCallback(() => goBack(), [goBack]);

  const slides: Slide[] = [
    {id: '1', text: 'hello', color: color.info},
    {id: '2', text: 'bob', color: color.warning},
    {id: '3', text: 'steve', color: color.success},
    {id: '4', text: 'jill', color: color.brand},
  ];

  return (
    <Screen onLeftPress={navBack} title="Landing">
      <Carousel slides={slides} />
      <Button center onPress={navPortfolioLogin} title="Login" />
    </Screen>
  );
});
