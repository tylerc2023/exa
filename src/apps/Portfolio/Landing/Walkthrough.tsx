import React, {memo, useCallback, useRef, useState} from 'react';
import {FlatList, StyleSheet, View} from 'react-native';
import {Text, TouchableOpacity} from '../../../components';
import {useColor, useDropShadow} from '../../../hooks';
import {getWidth} from '../../../models';
import {colorWithOpacity, Theme, useRootSelector} from '../../../utils';

interface Slide {
  id: string;
  text: string;
  color: string;
}

interface Props {
  dotSize?: number;
  viewabilityConfig?: Record<string, unknown>;
}

export const Walkthrough = memo(function Walkthrough({
  dotSize = Theme.padding.p04,
  viewabilityConfig = {itemVisiblePercentThreshold: 50},
}: Props) {
  const dropShadow = useDropShadow();
  const width = useRootSelector(getWidth);
  const color = useColor();
  const flatList = useRef<FlatList | null>(null);
  const [activeIndex, setActiveIndex] = useState(0);

  const onViewableItemsChanged = useCallback(({viewableItems}) => {
    setActiveIndex(viewableItems[0].index || 0);
  }, []);

  const slides: Slide[] = [
    {id: '1', text: 'hello', color: color.info},
    {id: '2', text: 'bob', color: color.warning},
    {id: '3', text: 'steve', color: color.success},
    {id: '4', text: 'jill', color: color.brand},
  ];
  const keyExtractor = useCallback((item) => String(item.id), []);
  const onDotPress = useCallback(
    (index: number) => () => {
      if (!flatList.current) {
        return;
      }

      flatList.current.scrollToIndex({index});
    },
    [],
  );
  const renderItem = useCallback(
    ({item}) => {
      const styles = StyleSheet.create({
        item: {
          backgroundColor: item.color,
          justifyContent: 'center',
          width,
        },
      });

      return (
        <View style={styles.item}>
          <Text center title={item.text} />
        </View>
      );
    },
    [width],
  );

  return (
    <View style={{flex: 1}}>
      <FlatList
        data={slides}
        horizontal
        keyExtractor={keyExtractor}
        onViewableItemsChanged={onViewableItemsChanged}
        pagingEnabled
        ref={flatList}
        renderItem={renderItem}
        viewabilityConfig={viewabilityConfig}
      />
      <View
        style={{
          position: 'absolute',
          flexDirection: 'row',
          bottom: 0,
          width: '100%',
          justifyContent: 'center',
        }}>
        {slides.map((slide, index) => {
          return (
            <TouchableOpacity
              key={slide.id}
              onPress={onDotPress(index)}
              style={{
                marginHorizontal: Theme.padding.p01,
                marginBottom: Theme.padding.p02,
                width: dotSize,
                height: dotSize,
                borderRadius: dotSize,
                borderColor: color.text,
                ...dropShadow(4),
                backgroundColor: colorWithOpacity(
                  activeIndex === index ? color.text : color.background,
                  0.6,
                ),
              }}
            />
          );
        })}
      </View>
    </View>
  );
});
