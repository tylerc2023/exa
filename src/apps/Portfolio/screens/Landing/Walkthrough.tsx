import React, { memo } from "react";
import { FlatList, StyleSheet, View } from "react-native";
import { Text } from "../../../../components";
import { useColor } from "../../../../hooks";
import { useRootSelector } from "../../../../utils";
import { getWidth } from "../../../../models";

interface Data {
  key: number;
  text: string;
  color: string;
}

const styles = (width: number, color: string) =>
  StyleSheet.create({
    item: {
      backgroundColor: color,
      justifyContent: "center",
      width
    }
  });

export default memo(function Walkthrough() {
  const width = useRootSelector(getWidth);
  const color = useColor();
  const data: Data[] = [
    { key: 1, text: "hello", color: color.primary },
    { key: 2, text: "bob", color: color.secondary },
    { key: 3, text: "steve", color: color.success },
    { key: 4, text: "jill", color: color.brand }
  ];

  const renderItem = ({ item }: { item: Data }) => 
    <View style={styles(width, item.color).item}>
      <Text title={item.text} center />
    </View>
  ;

  return (
    <FlatList
      pagingEnabled
      horizontal
      keyExtractor={item => String(item.key)}
      data={data}
      renderItem={renderItem}
    />
  );
});
