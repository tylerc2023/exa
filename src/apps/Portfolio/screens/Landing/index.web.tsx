import React, {memo} from 'react';
import {Text as OriginalText, ViewStyle, StyleProp} from 'react-native';
import {Text, TouchableOpacity, Icon} from '../../../../components';
import {View, Image} from 'react-native';
import {useColor} from '../../../../hooks';
import {Theme} from '../../../../utils';
import {FontType} from '../../../../components/Text/utils';

interface SignInButtonProps {
  onPress: () => void;
  icon: string;
  title: string;
}
const SignInButton = memo(function SignInButton({
  onPress,
  icon,
  title,
}: SignInButtonProps) {
  const color = useColor();
  return (
    <TouchableOpacity
      style={{
        alignItems: 'center',
        justifyContent: 'center',
        flexDirection: 'row',
        borderColor: color.text,
        borderWidth: 2,
        padding: Theme.padding.p02,
        marginBottom: Theme.padding.p02,
        borderRadius: Theme.padding.p10,
      }}
      onPress={onPress}>
      <Icon name={icon} style={{width: '30px', height: '30px'}} />
      <Text title={title} style={{paddingLeft: Theme.padding.p02}} type="h4" />
    </TouchableOpacity>
  );
});

interface NavButtonProps {
  onPress: () => void;
  inverted?: boolean;
  title: string;
}
const NavButton = memo(function NavButton({
  onPress,
  title,
  inverted,
}: NavButtonProps) {
  const color = useColor();
  return (
    <TouchableOpacity
      style={{
        justifyContent: 'center',
        borderRadius: Theme.padding.p10,
        paddingVertical: Theme.padding.p02,
        paddingHorizontal: Theme.padding.p02,
        backgroundColor: inverted ? color.text : color.background,
      }}
      onPress={onPress}>
      <Text title={title} type="h5" bold inverse={inverted} />
    </TouchableOpacity>
  );
});

type Word = {
  title: string;
  bold?: boolean;
  type?: FontType;
  functionality?: 'link' | 'text';
  onPress?: () => void;
};
interface SentenceProps {
  words: Word[];
  style: StyleProp<ViewStyle>;
}
const Sentence = memo(function Sentence({words, style}: SentenceProps) {
  return (
    <OriginalText style={style}>
      {words.map((word) => (
        <Text
          type={word.type}
          key={word.title}
          title={word.title}
          bold={word.bold}
          onPress={word.onPress}
        />
      ))}
    </OriginalText>
  );
});

const image = require('../../../../assets/line-chart.png');
export default memo(function PortfolioLanding() {
  const color = useColor();
  const titleSentence: Word[] = [
    {title: 'Get started with ', type: 'h2'},
    {title: 'Core', bold: true, type: 'h2'},
  ];
  const passwordSentence: Word[] = [
    {title: 'Or use your password to '},
    {
      title: 'sign up ',
      functionality: 'link',
      bold: true,
      onPress: () => undefined,
    },
    {title: 'or '},
    {
      title: 'sign in ',
      functionality: 'link',
      bold: true,
      onPress: () => undefined,
    },
  ];

  return (
    <View style={{height: '100vh'}}>
      <View
        style={{
          flexDirection: 'row',
          justifyContent: 'space-between',
          borderColor: color.secondary,
          padding: Theme.padding.p04,
          borderBottomWidth: 1,
        }}>
        <View style={{flexDirection: 'row', alignItems: 'center'}}>
          <Image source={image} style={{width: '30px', height: '30px'}} />
          <Text
            title="Core"
            type="h3"
            bold
            style={{paddingHorizontal: Theme.padding.p04}}
          />
          <NavButton title="Features" onPress={() => undefined} />
          <NavButton title="Premium" onPress={() => undefined} />
        </View>
        <View style={{flexDirection: 'row', justifyContent: 'flex-end'}}>
          <NavButton title="Sign In" onPress={() => undefined} />
          <NavButton inverted title="Sign Up" onPress={() => undefined} />
        </View>
      </View>
      <View style={{flex: 1}}>
        <Sentence
          style={{paddingVertical: Theme.padding.p08, alignSelf: 'center'}}
          words={titleSentence}
        />

        <View style={{width: '400px', alignSelf: 'center'}}>
          <SignInButton
            icon="google"
            title="Continue with Google"
            onPress={() => undefined}
          />
          <SignInButton
            icon="facebook"
            title="Continue with Facebook"
            onPress={() => undefined}
          />
          <Sentence
            style={{
              alignSelf: 'center',
              paddingVertical: Theme.padding.p04,
            }}
            words={passwordSentence}
          />
        </View>
        <View style={{flex: 1}} />
        <View
          style={{
            flexDirection: 'row',
            justifyContent: 'center',
            marginBottom: Theme.padding.p04,
          }}>
          <NavButton
            title="About us"
            onPress={() =>
              window.open('https://www.linkedin.com/in/ethanneff', '_blank')
            }
          />
          <NavButton
            title="Contact us"
            onPress={() => window.open('mailto:ethan.neff@eneff.com', '_blank')}
          />
          <NavButton title="Terms of Service" onPress={() => undefined} />
          <NavButton title="Privacy Policy" onPress={() => undefined} />
        </View>
      </View>
    </View>
  );
});
