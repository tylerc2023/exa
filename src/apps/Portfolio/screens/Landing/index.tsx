import React, { memo } from "react";
import { StyleSheet, View } from "react-native";
import { Button, Screen } from "../../../../components";
import Walkthrough from "./Walkthrough";
import { useNav } from "../../../../hooks";

const styles = StyleSheet.create({
  row: { flexDirection: "row", justifyContent: "space-around" }
});

export default memo(function PortfolioLanding() {
  const nav = useNav();
  return (
    <Screen>
      <Text h1 title="Landing" center />
      <Walkthrough />
      <View style={styles.row}>
        <Button title="Login" onPress={nav.to("portfolioLogin")} />
        <Button title="Main" onPress={nav.to("portfolio")} />
        <Button title="Settings" onPress={nav.to("portfolioSettings")} />
      </View>
      <View style={styles.row}>
        <Button title="Debug" onPress={nav.to("debug")} />
        <Button title="Checklist" onPress={nav.to("checklists")} />
        <Button title="CantHurtMe" onPress={nav.to("cantHurtMe")} />
        <Button title="Focus" onPress={nav.to("focus")} />
      </View>
    </Screen>
  );
});
