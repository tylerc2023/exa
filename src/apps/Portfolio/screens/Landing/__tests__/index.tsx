import * as React from "react";
import { Landing } from "../../Landing";
import { mockRenderer } from "../../../../../utils/Mock";

it("renders correctly", () => {
  const dom = mockRenderer(<Landing />).toJSON();
  expect(dom).toMatchSnapshot();
});
