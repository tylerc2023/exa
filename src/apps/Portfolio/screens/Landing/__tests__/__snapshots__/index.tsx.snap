// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "hsl(0, 0%, 100%)",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
          "padding": 12,
        }
      }
    />
    <View
      accessible={true}
      clickable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
          "padding": 12,
        }
      }
    />
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "hsl(0, 0%, 100%)",
          "flex": 1,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <Text
        style={
          Object {
            "color": "hsl(0, 0%, 0%)",
            "fontSize": 48,
            "fontWeight": "100",
            "letterSpacing": -1.5,
            "opacity": 1,
            "textAlign": "center",
          }
        }
      >
        Landing
      </Text>
      <RCTScrollView
        data={
          Array [
            Object {
              "color": "hsl(211, 100%, 50%)",
              "key": 1,
              "text": "hello",
            },
            Object {
              "color": "hsl(208, 7%, 46%)",
              "key": 2,
              "text": "bob",
            },
            Object {
              "color": "hsl(134, 61%, 41%)",
              "key": 3,
              "text": "steve",
            },
            Object {
              "color": "hsl(264, 34%, 36%)",
              "key": 4,
              "text": "jill",
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        pagingEnabled={true}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "hsl(211, 100%, 50%)",
                  "justifyContent": "center",
                  "width": 750,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "hsl(0, 0%, 0%)",
                    "fontSize": 14,
                    "fontWeight": "300",
                    "letterSpacing": 0.25,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                hello
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "hsl(208, 7%, 46%)",
                  "justifyContent": "center",
                  "width": 750,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "hsl(0, 0%, 0%)",
                    "fontSize": 14,
                    "fontWeight": "300",
                    "letterSpacing": 0.25,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                bob
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "hsl(134, 61%, 41%)",
                  "justifyContent": "center",
                  "width": 750,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "hsl(0, 0%, 0%)",
                    "fontSize": 14,
                    "fontWeight": "300",
                    "letterSpacing": 0.25,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                steve
              </Text>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "hsl(264, 34%, 36%)",
                  "justifyContent": "center",
                  "width": 750,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "hsl(0, 0%, 0%)",
                    "fontSize": 14,
                    "fontWeight": "300",
                    "letterSpacing": 0.25,
                    "opacity": 1,
                    "textAlign": "center",
                  }
                }
              >
                jill
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(211, 100%, 50%)",
                "fontSize": 14,
                "fontWeight": "600",
                "letterSpacing": 0.75,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            LOGIN
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(211, 100%, 50%)",
                "fontSize": 14,
                "fontWeight": "600",
                "letterSpacing": 0.75,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            MAIN
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(211, 100%, 50%)",
                "fontSize": 14,
                "fontWeight": "600",
                "letterSpacing": 0.75,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            SETTINGS
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(211, 100%, 50%)",
                "fontSize": 14,
                "fontWeight": "600",
                "letterSpacing": 0.75,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            DEBUG
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(211, 100%, 50%)",
                "fontSize": 14,
                "fontWeight": "600",
                "letterSpacing": 0.75,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            CHECKLIST
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(211, 100%, 50%)",
                "fontSize": 14,
                "fontWeight": "600",
                "letterSpacing": 0.75,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            CANTHURTME
          </Text>
        </View>
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 36,
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "hsl(211, 100%, 50%)",
                "fontSize": 14,
                "fontWeight": "600",
                "letterSpacing": 0.75,
                "opacity": 1,
                "textAlign": "center",
              }
            }
          >
            FOCUS
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
